fillin/src/main/java/com/teamfillin/fillin/domain/account/AccountService.java

```diff
@@ -0,0 +1,36 @@
+package com.teamfillin.fillin.domain.account;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class AccountService {
+	private final AccountUserFacade accountUserFacade;
+
+	public AccountService(AccountUserFacade accountUserFacade) {
+		this.accountUserFacade = accountUserFacade;
+	}
+
+	@Transactional
+	public AccountAccessResult loginOrJoin(@NotNull SocialType socialType, @NotNull String idKey) {
+		final AccountAndUserResult accountAndUser = accountUserFacade.retrieveAccountAndUserOrNull(socialType, idKey);
+		if (joined(accountAndUser)) {
+			return AccountAccessResult.login(accountAndUser);
+		}
+
+		final AccountAndUserResult joinResult;
+		try {
+			joinResult = accountUserFacade.registerAccountAndUser(socialType, idKey);
+		} catch (DataIntegrityViolationException e) { // unique error 에 대한 catch
+			final AccountAndUserResult foundAccountAndUser = accountUserFacade.retrieve(socialType, idKey);
+			return AccountAccessResult.login(foundAccountAndUser);
+		}
+		return AccountAccessResult.join(joinResult);
+	}
+
+	private boolean joined(AccountAndUserResult accountAndUser) {
```

Comment:
개인적으로 `isJoined` 가 좀 더 의미전달이나 가독성 측면에서 좋을 것 같아요
```suggestion
	private boolean isJoined(AccountAndUserResult accountAndUser) {
```

Reply:
제 개인 취향이긴한데요..ㅎ 수정해둘게용

Reply:
앗 저도 그냥 개인 생각이었습니다 ㅎㅎ 편하신대로 하셔도 될 것 같아요!